chronicle/src/main/java/net/openhft/chronicle/tcp/SinkTcp.java

```diff
@@ -65,7 +84,8 @@ public boolean open(boolean blocking) throws IOException {
                 socketChannel.socket().setSendBufferSize(this.builder.sendBufferSize());
             }
 
-            super.setSocketChannel(socketChannel);
+            this.socketChannel = socketChannel;
+            this.tcpEventHandler = new TcpEventHandler(socketChannel, builder.tcpPipeline(sinkTcpHandler), sessionDetailsProvider, builder.sendBufferSize(), builder.receiveBufferSize());
```

Comment:
It should be configurable, as today the default is blocking and the behavior is:
- if it is configured to be blocking, a call to nextIndex() will block until a message is received (heart-beat or data) and in such case it is not needed to spin on read 
- if it is configured as non blocking, nextIndex() will spins for a while on read then it gives up